pkg/proxy/userspace/proxier.go

```diff
@@ -373,23 +390,27 @@ func (proxier *Proxier) OnServiceUpdate(services []api.Service) {
 				glog.V(4).Infof("Something changed for service %q: stopping it", serviceName)
 				err := proxier.closePortal(serviceName, info)
 				if err != nil {
+					proxier.recorder.Eventf(proxier.nodeRef, "FailedaToClosePortal", "Failed to close portal for %q", serviceName)
 					glog.Errorf("Failed to close portal for %q: %v", serviceName, err)
 				}
 				err = proxier.stopProxy(serviceName, info)
 				if err != nil {
+					proxier.recorder.Eventf(proxier.nodeRef, "FailedToStopService", "Failed to stop service %q", serviceName)
 					glog.Errorf("Failed to stop service %q: %v", serviceName, err)
 				}
 			}
 
 			proxyPort, err := proxier.proxyPorts.AllocateNext()
 			if err != nil {
-				glog.Errorf("failed to allocate proxy port for service %q: %v", serviceName, err)
+				proxier.recorder.Eventf(proxier.nodeRef, "FailedToAllocateProxyPort", "Failed to allocate proxy port for service %q", serviceName)
+				glog.Errorf("Failed to allocate proxy port for service %q: %v", serviceName, err)
 				continue
 			}
 
 			glog.V(1).Infof("Adding new service %q at %s:%d/%s", serviceName, serviceIP, servicePort.Port, servicePort.Protocol)
 			info, err = proxier.addServiceOnPort(serviceName, servicePort.Protocol, proxyPort, udpIdleTimeout)
 			if err != nil {
+				proxier.recorder.Eventf(proxier.nodeRef, "FailedProxying", "Proxying failed for service %q", serviceName)
```

Comment:
The string here is wrong - we didn't fail to proxy, we failed to START the proxy.